### PR TITLE
Next cycle visible in support

### PR DESCRIPTION
**Pull request by inulty-dfe**
## Context

Admins and support users want to be able to see the data for the next cycle in the Support Interface. Right now they can only see up to the current cycle.

## Changes proposed in this pull request

- Add RSpec metadata hook for setting the HostingEnvironment in the test suite.
  - `describe 'some test', hosting_env: 'production' do`
- Change Support Interface course table heading from "Accredited body" to "Accredited provider"

### Screenshots

|With courses in next cycle| Without courses in next cycle|
|---|---|
|with-courses|without-courses|

Bonus: Some developer documentation around TTAPI

## Guidance to review

The refactoring of the Courses controller action and template could be improved, not sure how to make the controller action cleaner just now, but I feel it's better to push complexity up the stack (out of the view).

## Link to Trello card

[Preview next cycle data in SupportInterface](https://trello.com/c/3x2ac8YG/396-itt-reform-preview-2024-courses-in-manage)

## Things to check

- [ ] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] If this code adds a column to the DB, decide whether it needs to be in analytics yml file or analytics blocklist
- [ ] API release notes have been updated if necessary
- [ ] If it adds a significant user-facing change, is it documented in the [CHANGELOG](CHANGELOG.md)?
- [ ] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

File: app/controllers/support_interface/providers_controller.rb [hunk 1]
```diff
@@ -21,7 +21,11 @@ def show
 
     def courses
       @provider = Provider.includes(courses: [:accredited_provider]).find(params[:provider_id])
-      @courses = @provider.courses.includes(accredited_provider: [:provider_agreements]).order(:name).group_by(&:recruitment_cycle_year)
+
+      courses = @provider.courses.includes(accredited_provider: [:provider_agreements]).order(:name).group_by(&:recruitment_cycle_year)
+      years = RecruitmentCycle.years_visible_in_support.each_with_object({}) { |year, hash| hash[year] = [] } # rubocop:disable Rails/IndexWith
```
Comment (JR-G) [inline app/controllers/support_interface/providers_controller.rb:26]: Would we consider moving this into a private method?

**Review by JR-G — APPROVED** [review]
LGTM Iain ✅  Tested on the review app
image

File: app/controllers/support_interface/providers_controller.rb (hunk 1 above)
Comment (inulty-dfe) [reply app/controllers/support_interface/providers_controller.rb:26]: I wouldn't think it's necessary until it's behaviour that needs to be shared. The fact that it's in line in that action tells it's own story, that it is specific to that action. That's my point of view at least.

File: app/views/support_interface/providers/courses.html.erb [hunk 2]
```diff
@@ -5,11 +5,15 @@
 <% end %>
 
 <% if @provider.courses.any? %>
-  <% RecruitmentCycle.years_visible_in_support.each do |year| %>
-    <% if @courses[year] %>
-      <h2 class="govuk-heading-m"><%= year %>: <%= pluralize(@courses[year].size, 'course') %> (<%= @courses[year].count(&:open_on_apply?) %> on DfE Apply)</h2>
+  <% @courses_by_year.each do |year, courses| %>
+    <% if courses.any? %>
+      <h2 class="govuk-heading-m"><%= year %>: <%= pluralize(courses.size, 'course') %> (<%= courses.count(&:open_on_apply?) %> on DfE Apply) <%= 'Not yet published' if RecruitmentCycle.next_year == year %></h2>
```
Comment (Kateharries) [inline app/views/support_interface/providers/courses.html.erb:10]: 'not yet published' rather than 'Not yet published' - needs a lower case 'n'.

**Review by Kateharries — APPROVED** [review]
Looks good. I've added one comment.

**Pull request merged**